src/pynwb/__init__.py

```diff
@@ -225,6 +225,20 @@ def __init__(self, **kwargs):
                 manager = get_manager()
         super(NWBHDF5IO, self).__init__(path, manager=manager, mode=mode, file=file_obj, comm=comm)
 
+    @docval(*get_docval(_HDF5IO.read))
+    def read(self, **kwargs):
+        try:
+            return super(NWBHDF5IO, self).read(**kwargs)
+        except ValueError as e:
+            built = next(iter(self._HDF5IO__built.values()))
+            if '/nwb_version' in built and built['/nwb_version'].data[0] == '1':
+                raise Exception('file is version {} and is not supported by pynwb. See '
+                                'http://neurodatawithoutborders.github.io/api-python/ for help reading this file'
+                                ' (or just use h5py).'.
+                                format(built['/nwb_version'].data))
```

Comment:
I would not match for equality on the version here. It should be enough to check if `/nwb_version` is a dataset (it is an attribute for NWB v2). I've coded something along these lines in https://github.com/AllenInstitute/ipfx/blob/664f9f3578fe9a34f27c80b742b2af91459521c7/ipfx/nwb_reader.py#L606.

I would also not promote `http://neurodatawithoutborders.github.io/api-python/` and just mention h5py.

Reply:
I concur